rajawali/src/main/java/c/org/rajawali3d/core/ACoreComponent.java

```diff
@@ -0,0 +1,160 @@
+package c.org.rajawali3d.core;
+
+import c.org.rajawali3d.annotations.RenderThread;
+
+import android.support.annotation.CallSuper;
+import android.support.annotation.FloatRange;
+
+import net.jcip.annotations.GuardedBy;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Base class for all core components; enables common render thread tasking
+ *
+ * @author Randy Picolet
+ */
+
+abstract class ACoreComponent implements CoreComponent {
```

Comment:
I am generally not a fan of the `A` or `I` prefixes, in spite of the fact I am guilty of having used them before. This basically goes along with my desire to remove Hungarian Notation from the library.